test/issue-token.ts

```diff
@@ -0,0 +1,207 @@
+import * as Lab from "@hapi/lab";
+import { expect } from "@hapi/code";
+import nock from "nock";
+
+const lab = Lab.script();
+const { describe, it } = lab;
+export { lab };
+
+import { createInitializedServer } from "./util/server";
+import plugin from "../lib";
+
+type ScopeGeneratorOptions = { tokenEndpoint: string };
+const passwordGrantNockScope = ({ tokenEndpoint }: ScopeGeneratorOptions) => {
+  const tokenEndpointUrl = new URL(tokenEndpoint);
+  return nock(tokenEndpointUrl.origin)
+    .post(tokenEndpointUrl.pathname)
+    .reply(204, {
+      access_token: "ACCESS TOKEN",
+      expires_in: 3600,
+      token_type: "Bearer",
+      id_token: "ID TOKEN",
+      refresh_token: "REFRESH TOKEN",
+    });
+};
+
+const badPasswordGrantNockScope = ({
+  tokenEndpoint,
+}: ScopeGeneratorOptions) => {
+  const tokenEndpointUrl = new URL(tokenEndpoint);
+  return nock(tokenEndpointUrl.origin)
+    .post(tokenEndpointUrl.pathname)
+    .reply(400, function () {
+      // @ts-ignore bad types in nock
+      this.req.response.statusMessage = "Bad Request";
+      return {
+        error: "invalid_grant",
+        error_description: "invalid credentials provided",
+      };
+    });
+};
+
+const badGrantNockScope = ({ tokenEndpoint }: ScopeGeneratorOptions) => {
+  const tokenEndpointUrl = new URL(tokenEndpoint);
+  return nock(tokenEndpointUrl.origin)
+    .post(tokenEndpointUrl.pathname)
+    .reply(400, function () {
+      // @ts-ignore bad types in nock
+      this.req.response.statusMessage = "Bad Request";
+      return {
+        error: "unsupported_grant_type",
+        error_description: "unsupported grant_type requested (FAKE GRANT)",
+      };
+    });
+};
```

Comment:
These could be better organized, but all the options don't feel great. This file is short enough that it doesn't bother me to leave these here for now and defer that decision til later.